packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -83,6 +84,10 @@ function getAllMagicCommentDirectiveStyles(
       // Text uses HTML, front matter uses bash
       return getCommentPattern(['html', 'jsx', 'bash'], magicCommentDirectives);
 
+    case 'lua':
+    case 'haskell':
+      return getCommentPattern(['lua'], magicCommentDirectives);
+
     default:
       // All comment types
       return getCommentPattern(
```

Comment:
You should remove the `lua` pattern by default. There are too many false-positives

Reply:
You're right. I have updated the PR